### ISSUE TITLE
Parsed body 

### ISSUE THREAD

**Issue by Smolevich** [spiral/roadrunner#198]
We have this processing in client https://github.com/spiral/roadrunner/blob/85a2a735ea51252f6e9cf396b99e0e28f195f9da/src/PSR7Client.php#L100-L106
We got a problem with parsed body, when we send request with content type `application/json` and in php code we call `$request->getParsedBody()` and get null, but `$req->getBody()` return stream which can be used for reading post data
Is it expected behavior?

**Comment by wolfy-j** [comment]
Yes, RoadRunner does not parse any custom payload types (including JSON), this behavior is compliant with PSR-7. Use payload parsing middleware like this one: 
https://github.com/middlewares/payload
or this one
https://github.com/spiral/framework/blob/master/src/Http/Middleware/JsonPayloadMiddleware.php

**Comment by Smolevich** [comment]
For what case is this condition created? https://github.com/spiral/roadrunner/blob/85a2a735ea51252f6e9cf396b99e0e28f195f9da/src/PSR7Client.php#L100

**Comment by Smolevich** [comment]
I check docs and don't find info about examples using middlewares, this info can be added to docs in future?

**Comment by wolfy-j** [comment]
RoadRunner exposes PSR-7 interfaces to you, most of the frameworks implement their own middleware pipeline. This functionality not a part of RR but your application.

**Comment by Smolevich** [comment]
@wolfy-j , thanks for reply
